### PR TITLE
Fixes #49 - Add unit tests for github webhook handler and helpers

### LINKED ISSUE #49: Add unit testing and fixture files for GitHub webhooks handler

### PR DESCRIPTION
**Pull request by laghee**
Still working on this. There's lots of documentation on how to mock api/request data in python and much less on database operations. A lot of this may need to be figured out somewhat trial and error. 🤔  

Any suggestions or resource links for mocking/patching database operations more than welcome! (Any comments or suggestions on code-in-progress also appreciated.)

### CODE REVIEW

**Comment by laghee** [comment]
@karlcow OK! They might be ugly in places and can certainly be improved, but all 29 of these suckers now pass locally.
(The test file is really long. Please don't kill me... 😜   Could also be useful as a soporific if you ever suffer from insomnia!)

**Comment by miketaylr** [comment]
@laghee do you know what's up with the following failure?

```
...........E
======================================================================
ERROR: tests.unit.test_webhooks (nose2.loader.ModuleImportFailure)
----------------------------------------------------------------------
ImportError: Failed to import test module: tests.unit.test_webhooks
Traceback (most recent call last):
  File "/home/circleci/repo/venv/lib/python3.6/site-packages/nose2/plugins/loader/discovery.py", line 201, in _find_tests_in_file
    module = util.module_from_name(module_name)
  File "/home/circleci/repo/venv/lib/python3.6/site-packages/nose2/util.py", line 77, in module_from_name
    __import__(name)
  File "/home/circleci/repo/tests/unit/test_webhooks.py", line 21, in <module>
    from ochazuke.webhook import helpers
ModuleNotFoundError: No module named 'ochazuke.webhook'
```

**Comment by karlcow** [comment]
@laghee @miketaylr  
> But they depend on each other, so I'm not sure how to make that work. I think I might need some git advice...

I don't think git will help that much at this stage. It's not the modular way I was thinking about when I was thinking about smaller PRs. 😃 but that's my bad for making it not clearer. 

Right now the code is split in big interdependent PRs organized by logical code organization. What I was hinting at was organizing PRs by little functional issues. Let's say for example handling the HTTP response and its tests. Then going to the next stage in processing data and its tests. etc. :)

All of that said we will survive. Be a big unique PR or merging them one by one it will take time. We can't really review the tests currently because they are dependent on the rest of the code. So we have to go through others first. I'm planning to work on this a bit today.

**Comment by karlcow** [comment]
@laghee could you resolved the conflicting files by rebasing on the latest master.

File: tests/unit/test_webhooks.py [hunk 1]
```diff
@@ -0,0 +1,483 @@
...
+        # binds app to the current context
+        self.app_context = self.app.app_context()
+        self.app_context.push()
+        # create all the tables for testing
+        db.create_all()
+        self.client = self.app.test_client()
+        self.headers = {'content-type': 'application/json'}
+        self.test_url = '/webhooks/ghevents'
+        self.info = {'issue_id': 2475,
+                     'title': 'Cannot log in to www.artisanalmustard.com!',
```
[the remark this answers is not in the document]
Comment (laghee) [reply tests/unit/test_webhooks.py]: Hah! I should have figured. Freaking hipsters. 🤣

File: ochazuke/__init__.py [hunk 2]
```diff
@@ -40,7 +40,13 @@ def create_app(test_config=None):
         pass
 
     # configure the postgresql database
-    app.config['SQLALCHEMY_DATABASE_URI'] = os.environ.get('DATABASE_URL')
+    if test_config is None:
+        # fetch the environmental variable for the database location
+        database_url = os.environ.get('DATABASE_URL')
+    else:
+        # use the local database for testing
+        database_url = 'postgresql://localhost/metrics'
```
[the remark this answers is not in the document]
Comment (laghee) [reply ochazuke/__init__.py]: Once we get to integration testing, yes. We could wait to add this, though. (I initially thought I might go right to integration when I was really struggling with mocking.)

File: tests/fixtures/webhooks/issue_event_invalid_action.json [hunk 3]
```diff
@@ -0,0 +1,44 @@
+{
+    "action": "ochazuked",
```
[the remark this answers is not in the document]
Comment (laghee) [reply tests/fixtures/webhooks/issue_event_invalid_action.json]: Heh. I hoped you'd enjoy that. I'll bet many of us would benefit from being ochazuked more often. 😋

**Comment by laghee** [comment]
> Right now the code is split in big interdependent PRs organized by logical code organization. What I was hinting at was organizing PRs by little functional issues. Let's say for example handling the HTTP response and its tests. Then going to the next stage in processing data and its tests. etc. :)
> 
Ahhh. Ooops. I get it now.

> All of that said we will survive. Be a big unique PR or merging them one by one it will take time. We can't really review the tests currently because they are dependent on the rest of the code. So we have to go through others first.

I'll make the `artisanalmustard.example.com` change and rebase for now.

**Comment by laghee** [comment]
@karlcow OK, changes made. I also took out the `db` statements since they're not needed with mocking and they were freaking out CircleCI. (Looks like the setup I added for accessing the postgres db in Circle isn't quite right (issue #57), which I'll need to figure out when we start working on integration testing).

**Comment by karlcow** [comment]
Thanks a lot @laghee ❤️

**Pull request merged**